### PR TITLE
Fix/chain selector tabs

### PR DESCRIPTION
**Pull request by kevinhcolgan**
It's not a nice UX if we always render tabs in the chain selector i.e.: 
image

This PR adds logic to only render tabs if they are necessary, i.e. if there are a mix of testnet and mainnet networks, otherwise, just render a div without tabs:
chain_selector_test_mainnet_tabs

### CODE REVIEW

File: packages/core/src/components/ChainSelector.tsx [hunk 1]
```diff
@@ -180,53 +180,67 @@ export function ChainSelectorContent<
...
   const { labels } = useLabel();
 
-  const [hasTestnetChains, setHasTestnetChains] = useState<boolean>(false);
-
-  useEffect(() => {
-    if (chains.filter((chain) => chain.testnet === true)?.length >= 1) {
-      setHasTestnetChains(true);
-    }
+  const hasTestnetChains = useMemo(() => {
+    return chains.filter((chain) => chain.testnet === true)?.length >= 1;
```
Comment (flipscholtz) [inline packages/core/src/components/ChainSelector.tsx:184]: `return` can be removed

File: packages/core/src/components/ChainSelector.tsx [hunk 2]
```diff
@@ -180,53 +180,67 @@ export function ChainSelectorContent<
...
+  }, [chains]);
+  const hasMainnetChains = useMemo(() => {
+    return chains.filter((chain) => chain.testnet !== true)?.length >= 1;
```
Comment (flipscholtz) [inline packages/core/src/components/ChainSelector.tsx:187]: `return` can be removed

File: packages/examples/advanced-react/src/App.tsx [hunk 3]
```diff
@@ -68,8 +68,8 @@ function Content() {
 }
 
 function App() {
-  const defaultEvmChains = [polygon, mainnet, arbitrum];
-  const defaultEvmTestChains = [polygonMumbai, goerli, arbitrumGoerli];
+  const defaultEvmChains: EthereumChain[] = [polygon, mainnet, arbitrum];
+  const defaultEvmTestChains: EthereumChain[] = [];
```
Comment (flipscholtz) [inline packages/examples/advanced-react/src/App.tsx:72]: Why no more default EVM test chains?

Comment (kevinhcolgan) [reply packages/examples/advanced-react/src/App.tsx:72]: oops, that was mistake

File: packages/core/src/components/ChainSelector.tsx (hunk 2 above)
Comment (kevinhcolgan) [reply packages/core/src/components/ChainSelector.tsx]: fixed

File: packages/core/src/components/ChainSelector.tsx (hunk 1 above)
Comment (kevinhcolgan) [reply packages/core/src/components/ChainSelector.tsx]: fixed

File: packages/examples/advanced-react/src/App.tsx (hunk 3 above)
Comment (kevinhcolgan) [reply packages/examples/advanced-react/src/App.tsx]: fixed

**Review by TYRONEMICHAEL — APPROVED** [review]

**Pull request merged**